File: backend/app/api/logs.py
from fastapi import APIRouter, Query

from app.services.logs.runtime_buffer import runtime_log_buffer

router = APIRouter()


@router.get("/recent")
def get_recent_logs(
    source: str = Query("all", pattern="^(task|service|all)$"),
    limit: int = Query(200, ge=1, le=500),
    since: str | None = None,
):
    items = runtime_log_buffer.list_recent(source=source, since=since, limit=limit)
    next_since = items[-1]["timestamp"] if items else since
    return {"items": items, "next_since": next_since}

File: frontend/src/api/modules/logs.ts
import http from '@/api/http'
import type { LogsResponse } from '@/types'

export async function fetchRecentLogs(params?: {
  source?: 'task' | 'service' | 'all'
  limit?: number
  since?: string
}) {
  const { data } = await http.get<LogsResponse>('/api/v1/logs/recent', { params })
  return data
}
